### ISSUE TITLE
Match reported performances

### ISSUE THREAD

**Issue by juanmirocks** [Rostlab/relna#21]
## Reported:
- relna (test set): `F 69.3%`
- LocText (validation set): `P 66.57%, R 67.61%, F 66.73% (with DS)` -- `63.22 (just SS)`
## August 4th 2016
- relna (_with_ class balancing, no SVM TK, ~~no hyperparameter tuning~~): `F 65.55%` -- `F 65.10% (with TK)`
- LocText (using relna features and no DS model, same as before): `F 58.05 %`
## October 27th 2016
- [x] relna (**dev** set, _with_ class balancing, no SVM TK, cv-5): `F 0.7047` 😀

```
# class tp  fp  fn  fp_ov   fn_ov   e|P e|R e|F e|F_SE  o|P o|R o|F o|F_SE
r_4 179 82  68  0   0   0.6858  0.7247  0.7047  0.0019  0.6858  0.7247  0.7047  0.0019
```
- relna on **test** set:

```
# class tp  fp  fn  fp_ov   fn_ov   e|P e|R e|F e|F_SE  o|P o|R o|F o|F_SE
r_4 186 92  61  0   0   0.6691  0.7530  0.7086  0.0018  0.6691  0.7530  0.7086  0.0020
```

---
# 🚒 ⚠️ 🤔 IMPORTANT 🤔 ⚠️ 🚒

While investigating relna's code, I (Juanmi), saw that Ashish had established the threshold of svm_lights to `-0.1` instead of the default/supposedly-correct `0`. See relna commit (e682d8af63e92f044ea41dee5b90e8a3033795cb) and nalaf commit (https://github.com/Rostlab/nalaf/commit/f7250c503f3d1b1c05f14391146f3c56df7e86cf). There is no explanation for this in Ashish's master thesis.

`-0.1` actually performs **BETTER** than `0`:
- dev set:
  - `0`: 0.6931
  - `-0.1`: 0.7008
- test set:
  - `0`: 0.6889
  - `-0.1`: 0.7047

This seems to be **ABOVE standard error**. Therefore, as of now:
1. we define the threshold to be a parameter of svm_light (https://github.com/Rostlab/nalaf/commit/08f9995d5dc3db0f3c40d5774dec2e1c209e279c) 
2. we set the threshold to `-0.1` in relna, for no other reason that so far it seems to perform better (e3093eb191cdb09fcd6433147ce1186a4f674793)

A priori, it seems reasonable that performance may improve by setting a lower value in the threshold (which increases the number of instances predictive _positively_, thus oversampling that class, which is in the first place undersampled)

As a matter of fact, the **same behavior is observed with the LocText corpus**

---

⚠️ 

**Still, this whole thing is fishy fishy, and I am unaware that Ashish Baghudana did set this value with a thought-out decision**. There is no info in his thesis: https://github.com/Rostlab/relna/files/311536/Ashish.Baghudana_bits_thesis.pdf

---
# Even more ‼️

The current used parser is just `spacy` (_without using the `bllip parser`_), and thus the tokens' ~~dependency/~~constituency features are not added. This renders useless ~~the~~ some features of at least the generators: `PathFeatureGenerator` and `ProteinWordFeatureGenerator`

~~and yet~~ the current performance is even higher than the one reported by Ashish 🤔 💭

**Comment by juanmirocks** [comment]
Close issue here since problem on _relna_ is solved. The following up on LocText is described in: https://github.com/juanmirocks/LocText/issues/11
